vms/platformvm/add_subnet_validator_tx_test.go

```diff
@@ -170,33 +182,27 @@ func TestAddSubnetValidatorTxExecute(t *testing.T) {
 		defaultWeight,
 		uint64(defaultValidateStartTime.Unix()+1),
 		uint64(defaultValidateEndTime.Unix()),
-		nodeID,
+		nodeIDs[0],
 		testSubnet1.ID(),
-		[]*crypto.PrivateKeySECP256K1R{testSubnet1ControlKeys[0], testSubnet1ControlKeys[1]},
+		[]*crypto.PrivateKeySECP256K1R{testSubnet1ControlKeys[0], testSubnet1ControlKeys[1], nodeKeys[0]},
 	); err != nil {
 		t.Fatal(err)
 	} else if _, _, err = tx.UnsignedTx.(UnsignedProposalTx).Execute(vm, vm.internalState, tx); err != nil {
 		t.Fatal(err)
 	}
 
-	// Add a validator to pending validator set of primary network
-	key, err := vm.factory.NewPrivateKey()
-	if err != nil {
-		t.Fatal(err)
-	}
-	pendingDSValidatorID := key.PublicKey().Address()
+	nodeKey, nodeID := generateNodeKeyAndID()
 
 	// starts validating primary network 10 seconds after genesis
 	DSStartTime := defaultGenesisTime.Add(10 * time.Second)
 	DSEndTime := DSStartTime.Add(5 * defaultMinStakingDuration)
 
 	addDSTx, err := vm.newAddValidatorTx(
```

Comment:
As mentioned before, nodeSig should NOT be any special case, it even should not be a new parameter passed to newAddValidatorTx